### PR TITLE
 Bug 1594443 -  Migrate graphics-telemetry notebooks scheduled in Databricks to read from BigQuery

### PR DESCRIPTION
**Pull request by acmiyaguchi**
This updates the `graphics-telemetry-dashboard` and `graphics-telemetry-trends` notebooks to reference code in the `analyses/bigquery_shim` package. The code mostly packages [these forked notebooks](https://bugzilla.mozilla.org/show_bug.cgi?id=1594443#c5).

The exported databricks notebooks end up being squashed into a single line and are nearly impossible to diff. I've added a script in `tools/notebook_to_source.sh` that can be used to dump notebooks into a `analyses/src` notebooks for review.

The two commits with the majority of the changes are  84de3b1 (for the dashboard) and 
3938f66 (for trends).

Updates to the library can be made by changing directories into `analyses/bigquery_shim` and running the `deploy.sh` script. This requires having the `dbfs` command-line tool set up to copy the egg into an appropriate location.

### CODE REVIEW

**Comment by mikokm** [comment]
Amazing, thank you for working on this. This version looks way more production ready than the previous one.
It's a big refactor, but I'll try to get this reviewed and deployed this week.

**Comment by mikokm** [comment]
Running these notebooks on DataBricks will yield a warning:
```
WARNING: python_moztelemetry will stop working *VERY SOON* (if it hasn't stopped working already)
Please see this post to fx-data-dev for more information: https://mail.mozilla.org/pipermail/fx-data-dev/2019-November/000291.html
```

and deprecation errors for `get_one_ping_per_client` and `get_pings_properties`.
After a quick look at the source code for these, it seems that they are just processing already fetched pings.
The question is, will importing these functions from moztelemetry just stop working at some point?

**Comment by acmiyaguchi** [comment]
The deprecation errors refer to the deletion of the underlying infrastructure on AWS which relied on the `moztelemetry.Dataset` api. The utility functions will continue to work, but they're not recommended for general use anymore. I would try to push the `get_one_ping_per_client` and `get_pings_properties` logic into SQL for any new notebooks/queries.

File: analyses/bigquery_shim/bigquery_shim/snake_case.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+    # filter spaces between words and snake_case and reverse again
+    return "_".join([w.lower() for w in words if w.strip()])[::-1]
+
+
+def split_snake_case(line):
+    return "/".join(list(map(snake_case, line.split("/"))))
+
+
+class SnakeCaseDictOld(dict):
```
Comment (whd) [inline analyses/bigquery_shim/bigquery_shim/snake_case.py:36]: This can be deleted.

**Review by whd — APPROVED** [review]
>  I would try to push the get_one_ping_per_client and get_pings_properties logic into SQL for any new notebooks/queries.

I agree with this. In the case of these rewritten queries we're doing both of these at the BQ level. `get_one_ping_per_client` should in theory be totally unnecessary when pushed up to BQ but the output of `get_pings_properties` does some histogram merging and conversion to pandas series, so for compatibility with moztelemetry you will want to still call it.

**Comment by mikokm** [comment]
I deployed this on Databricks and it works! There was an issue with stale data (the sample timestamp was from mid-October), but it magically worked a couple of hours later, so it might have just been a caching issue in S3.

**Review by mikokm — APPROVED** [review]

**Pull request merged**